src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -335,9 +335,20 @@ case class CsvRelation protected[spark] (
       }
       // Write the data. We assume that schema isn't changed, and we won't update it.
 
-      val codecClass = CompressionCodecs.getCodecClass(codec)
-      data.saveAsCsvFile(filesystemPath.toString, Map("delimiter" -> delimiter.toString),
-        codecClass)
+      val hadoopConf = data.sqlContext.sparkContext.hadoopConfiguration
+      val maybeCodecClass = CompressionCodecs.getCodecClass(codec)
+      val parameters = Map("delimiter" -> delimiter.toString)
+      maybeCodecClass match {
```

Comment:
I meant to explicitly set non-compression with ignoring global configuration so that it can be ignored by setting `compress` to `none` , even if the compression configuration is set to `sqlContext.sparkContext.hadoopConfiguration`.

Sorry, this would change the global configuration. Let me correct this.